datumaro/components/hl_ops.py

```diff
@@ -0,0 +1,192 @@
+# Copyright (C) 2022 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+from typing import Dict, Optional, Union
+import os
+import os.path as osp
+import shutil
+
+from datumaro.components.converter import Converter
+from datumaro.components.dataset import (
+    Dataset, DatasetItemStorageDatasetView, IDataset,
+)
+from datumaro.components.dataset_filter import (
+    XPathAnnotationsFilter, XPathDatasetFilter,
+)
+from datumaro.components.environment import Environment
+from datumaro.components.extractor import Transform
+from datumaro.components.launcher import Launcher, ModelTransform
+from datumaro.components.operations import ExactMerge
+from datumaro.components.validator import TaskType, Validator
+from datumaro.util import parse_str_enum_value
+from datumaro.util.scope import on_error_do, scoped
+
+
+def transform(dataset: IDataset, method: Union[str, Transform], *,
+        env: Optional[Environment] = None, **kwargs) -> IDataset:
+    """
+    Applies some function to dataset items.
+
+    Results are computed lazily, if the transform supports this.
+
+    Args:
+        dataset (IDataset) - The dataset to be transformed
+        method (str or Transform) - The transformation function to be
+            applied to the dataset. If a string is passed, it is treated
+            as a plugin name, which is searched for in the environment
+            set by the 'env' argument
+        env (Environment) - A plugin collection. If not set, the default one
+            is used
+        **kwargs - Optional parameters for the transformation
+
+    Returns: a lazy-computible wrapper around the input dataset
```

Comment:
"Computible" is not a word, but even if you replace it with "computable" I don't think this sentence would work well. The wrapper itself is computed eagerly, it's the items that aren't. IMO, it should be rephrased to something like "a wrapper around the input dataset whose items are determined lazily".

Reply:
Rephrased